packages/flutter/lib/src/rendering/binding.dart

```diff
@@ -215,7 +215,11 @@ mixin RendererBinding on BindingBase, ServicesBinding, SchedulerBinding, Gesture
       _debugIsRenderViewInitialized = true;
       return true;
     }());
+    assert(!_hasDeviceMetrics);
```

Comment:
I do'nt think this is safe. _hasDeviceMetrics may have gotten set in handleMetricsChanged before this.